bokeh/models/actions.py

```diff
@@ -15,3 +15,25 @@ class OpenURL(Action):
     The URL to direct the web browser to. This can be a template string,
     which will be formatted with data from the data source.
     """)
+
+class Callback(Action):
+    """ Execute a JavaScript function. """
+
+    args = Dict(String, Instance(PlotObject), help="""
+    A mapping of names to Bokeh plot obejcts. These objects are made
+    available to the callback code snippet as the values of named
+    parameters to the callback.
+    """)
+
+    code = String(help="""
+    A snippet of JavaScript code to execute in the browser. The code is
+    made into the body of a function, and all of of the named objects in
+    ``args`` are available as parameters that the code can use. Additionally,
+    a ``cb_obj`` parameter contains the object that triggered the callback.
+
+    .. warning:
+        Due to JS context and scoping rules, if the code snippet calls
+        external functions that use non-"var" variables with names that
+        conflict with names in ``args``, there is the possibility of subtle
+        errors.
```

Comment:
Well hell.  I just wrote up some test cases to verify this and it looks like I'm wrong.  I've read most of [MDN][1]'s docs on functions and scope this afternoon and I can't find references to the behavior I've described.  I know I've dealt with issues in the past where variables that were not bound to a particular scope were infected by the calling scope, but I can't for the life of me find a way to reproduce it on Chrome, Firefox, or io.js.

Best I can tell the function with a variable not bound to its scope will leak to to global, but a variable defined within a function (as an argument or a `var`) won't be effected by that leaking in subsequent calls.  Consider this example:

```javascript
var foo = function() {
  source = "modified by foo";
  console.log(source);
}

var bar = function() {
  var source = "set by bar";
  console.log(source);
  foo();
  console.log(source);
}

bar();
console.log("source: " + source);
```

In this case, `source` is *always* "set by bar" inside `bar()`, but changes inside `foo` and the global scope.

I would have lost money on this. *shakes head*

[1]: https://developer.mozilla.org/en-US/docs/Web